### DISCUSSION TITLE
What's left before having a complete hot-update solution

### DISCUSSION THREAD

**Discussion by fzyzcjy** [ethanblake4/dart_eval#73] [General]
IMHO (hopefully I am not too optimistic about this!) with the open source library of you and me, maybe we can create a full hot-update solution and solve the https://github.com/flutter/flutter/issues/14330.

I guess maybe we need to finish these components:

* dart JIT mode - use in Android (since faster)
* dart_eval and flutter_eval - use in ios, because it needs interpreted code
* flutter_smooth - use to speed up JIT and interpreted code
* glue and scaffolding - feed app code into JIT/dart_eval, etc

Anyway this issue is not a real "issue" (problem), but just a little reminder about what I think :)

**Comment by gintominto5329** [comment]
> dart JIT mode - use in Android (since faster)

I think, a better solution would be to compile some basic components, like `void main`, `class ScreeningPage`, and `class HomePage`, as AOT, rest all as JIT,

Because when flutter/dart team when asked regarding JIT-ing flutter apps, 
told that JIT slows down the app's start/initialization to at-least a few(2 or 4, if I remember correctly) seconds, due to its on-the-fly compilation,

thanks

**Comment by gintominto5329** [comment]


**Reply by gintominto5329** [reply]
@fzyzcjy, once you have read this, please intimate me

**Reply by fzyzcjy** [reply]
Hi, I am not the implementor of this library. Instead, ethanblake4 implements the code :)

**Comment by gintominto5329** [comment]


**Reply by maks** [reply]
@gintominto5329 it may be helpful for you to start with stating what you are actually trying to achieve and then it would be possible to assess if dart_eval (or some other technology or technique) would work to meet your goals.

**Comment by Zverik** [comment]
There is a hot update demo included with `flutter_eval`, and I guess this topic is a bit obsolete: with the current maintainance mode, it's unlikely the library will reach the production state in the visible future.
